### ISSUE TITLE
`no-unnecessary-condition` false positive with `true` in while loop test

### ISSUE THREAD

**Issue by gameroman** [oxc-project/tsgolint#877]
```ts
function fn() {
  while (true) { // Unnecessary conditional, value is always truthy.
    // ...

    if (condition) break
  }
}
```

**Comment by camc314** [comment]
Please configure the rule to allow loops (`allowConstantLoopConditions`)

https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-unnecessary-condition.html#allowconstantloopconditions

```
{
  "rules": {
    "typescript/no-unnecessary-condition": [
      "error",
      {
        "allowConstantLoopConditions": true
      }
      ]
  }
}
```

**Comment by gameroman** [comment]
Sorry, missed that

Thank you 👍
